src/bin/rbw-agent/actions.rs

```diff
@@ -277,6 +305,69 @@ async fn two_factor(
     unreachable!()
 }
 
+async fn yubikey_auth(
```

Comment:
Is there a reason why you chose to add an extra yubikey method instead of re-using `two_factor` which seems to be intended to be made more flexible since it takes the authentication mechanism as a parameter?

Reply:
There were a couple of reasons I decided on a separate function.

1. The message in the window should be different for YubiKey (but arguably you could also just `let message = match provider { ....`
1. The error handling is very subtly different. For example, this case https://github.com/dave-tucker/rbw/blob/290b0a9a511e9dfae91ebc0ffee12203c5a6cda5/src/bin/rbw-agent/actions.rs#L286 doesn't apply to YubiKey auth. I have a feeling - not verified - that the error codes from the API will be ever so slightly different for each auth mechanism and that we'll have to handle those separately.
1. I started thinking ahead to about how to implement webauthn support, and I concluded it would need it's own function since it would have to launch a browser window...

Reply:
I've played around with matching the provider and then setting the messages according to that, works fine, but is not really pretty.
We could take this further and implement a get_messages fn or something for the variant to isolate the match inside of the enum. 
Same could be done for validating pin input and error handling - that would keep the code in `two_factor` fairly clean and isolate handling of specific things in the enum.

Something along these lines:

```
impl TwoFactorProviderType {
    pub fn validate_entry(&self, entry: &str) -> anyhow::Result<()> {
        match self {
            TwoFactorProviderType::Authenticator => {Ok(())},
            TwoFactorProviderType::Yubikey => {Ok(())},
            unsupported_type => {Err(anyhow::anyhow!(
                "unsupported two factor authentication method encountered: {:?} (this should not happen, please consider raising an issue for this!)",unsupported_method
            ))}
        }
    }
}
```